File: leetcode/js/number-of-students-unable-to-eat-lunch.js
/**
 * @param {number[]} students
 * @param {number[]} sandwiches
 * @return {number}
 */
var countStudents = function (students, sandwiches) {
  sandwiches.reverse();

  while (students.length > 0) {
    const student = students.shift();

    if (sandwiches.at(-1) !== student) {
      students.push(student);

      if (students.every((student) => student !== sandwiches.at(-1))) {
        break;
      }
    } else {
      sandwiches.pop();
    }
  }

  return students.length;
};

File: leetcode/rust/number-of-students-unable-to-eat-lunch.rs
impl Solution {
    pub fn count_students(students: Vec<i32>, sandwiches: Vec<i32>) -> i32 {
        let mut count_students = [0, 0];

        for student in students {
            let num = student as usize;

            count_students[num] += 1;
        }

        for sandwich in sandwiches {
            let num = sandwich as usize;

            if (count_students[num] > 0) {
                count_students[num] -= 1;
            } else {
                break;
            }
        }

        count_students.iter().sum()
    }
}
